### PR TITLE
Enhancement/42 update volume context and volume mute controller

### PR DESCRIPTION
**Pull request by Prom1s1ngYoung**
### Descriptions
This demo is available in http://localhost:3000/volume, I create this demo in the main section, you can try that.

- Link to UI design (if possible):
No CSS style is needed now.

### Provide screenshots or videos that the function is working.

https://github.com/taodemy/taotify/assets/110534329/3c6613ce-f5dc-43d1-85ec-4fd53bd26f9b

### Provide screenshots of no warning, or error in the console.
image

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

**Review by tatoMa — COMMENTED** [review]
Creating separate VolumeContext fils is good for file management but may I ask how you plan to integrate with play music and other related function?

**Comment by Prom1s1ngYoung** [comment]
> 

There will be another web audio context, I can directly get the GainNode which can control the volume, and when the volume changes, I can directly set the volume to the current GainNode. Oppositely, when the audio source changes, there will be another new GainNode, when generating a new GainNode (hook state changed), the volume should also be set to the current GainNode.

File: src/contexts/VolumeContext.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  setVolumeLevel: React.Dispatch<React.SetStateAction<number>>;
+  backtrackVolumeLevel: number;
+  setBacktrackVolumeLevel: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const VolumeContext = createContext<VolumeContextType>({
+  volumeLevel: 50,
+  setVolumeLevel: () => {},
+  backtrackVolumeLevel: 50,
+  setBacktrackVolumeLevel: () => {},
```
[the remark this answers is not in the document]
Comment (Prom1s1ngYoung) [reply src/contexts/VolumeContext.tsx:14]: I will create a constant file that describes the default value that will allow us to control the default value easily. And we can temporarily set it to 50.

File: src/contexts/MusicContext.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { createContext, useState } from "react";
+import React, { createContext, useEffect, useState } from "react";
```
[the remark this answers is not in the document]
Comment (Prom1s1ngYoung) [reply src/contexts/MusicContext.tsx:1]: Sure.

File: src/components/volumeController/VolumeMuteController.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export default function VolumeMuteController() {
+  const { volumeLevel, setVolumeLevel, backtrackVolumeLevel } = useContext(VolumeContext);
+  const handleClick = () => {
+    if (volumeLevel > 0 && volumeLevel <= 100) {
+      setVolumeLevel(0);
+      return;
+    }
+    if (volumeLevel === 0) {
+      setVolumeLevel(backtrackVolumeLevel);
+      return;
```
[the remark this answers is not in the document]
Comment (Prom1s1ngYoung) [reply src/components/volumeController/VolumeMuteController.tsx:13]: Sure.

File: src/components/volumeController/VolumeController.tsx
[the remark this answers is not in the document]
Comment (Prom1s1ngYoung) [reply src/components/volumeController/VolumeController.tsx:1]: Sure.

File: src/contexts/VolumeContext.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+interface VolumeContextType {
+  volumeLevel: number;
+  setVolumeLevel: React.Dispatch<React.SetStateAction<number>>;
+  preMuteVolumeLevel: number;
+  setPreMuteVolumeLevel: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const VolumeContext = createContext<VolumeContextType>({
+  volumeLevel: defaultVolume,
+  setVolumeLevel: () => {},
```
Comment (tatoMa) [inline src/contexts/VolumeContext.tsx:13]: Do you think the maximum volume (100) condition should be placed here or create another method to do this?

File: src/constant/volume.ts
Comment (tatoMa) [inline src/constant/volume.ts:1]: Might be better to consider using capitalize the name of const.

Comment (Prom1s1ngYoung) [reply src/constant/volume.ts:1]: Ok

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request merged**